crates/ruma-events/src/secret/request.rs

```diff
@@ -0,0 +1,302 @@
+//! Types for the *m.secret.request* event.
+
+use std::convert::TryFrom;
+
+use ruma_events_macros::EventContent;
+use ruma_identifiers::DeviceIdBox;
+use ruma_serde::StringEnum;
+use serde::{ser::SerializeStruct, Deserialize, Serialize};
+
+use crate::ToDeviceEvent;
+
+/// Event sent by a client to request a secret from another device or to cancel a previous request.
+///
+/// It is sent as an unencrypted to-device event.
+pub type RequestToDeviceEvent = ToDeviceEvent<RequestToDeviceEventContent>;
+
+/// The payload for RequestToDeviceEvent.
+#[derive(Clone, Debug, Serialize, Deserialize, EventContent)]
+#[cfg_attr(not(feature = "unstable-exhaustive-types"), non_exhaustive)]
+#[ruma_event(type = "m.secret.request", kind = ToDevice)]
+pub struct RequestToDeviceEventContent {
+    /// The action for the request, one of `["request", "request_cancellation"]`.
```

Comment:
```suggestion
    /// The action for the request.
```

The possible values are documented in the `RequestAction` type's docs. Listing them here just increases the likelyhood of these docs becoming out-of-date.